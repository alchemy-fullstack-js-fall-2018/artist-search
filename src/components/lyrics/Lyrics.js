
import React, { Component } from 'react';
import { getLyrics } from '../../services/musicSearch';

export default class Lyrics extends Component {
  state = {
    artist: this.props.artist,
    song: this.props.song,
    lyrics: ''
  };

  fetchLyrics = () => {
    getLyrics(this.props.match.params.artist, this.props.match.params.song)
      .then(({ lyrics }) => this.setState({ lyrics }));
  };

  componentDidMount = () => {
    this.fetchLyrics();
  };

  render() {
    const { lyrics } = this.state;

    return (
      <p>{lyrics || 'Loading...'}</p>
    );
  }
}
