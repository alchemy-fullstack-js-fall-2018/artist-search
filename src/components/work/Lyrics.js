import React, { Fragment, Component } from 'react';
import { getLyrics } from '../../services/artistsAPI';

export default class Lyrics extends Component {
  state = {
    lyrics: ''
  };

  fetchLyrics = () => {
    const { artist, title } = this.props.match.params;
    getLyrics(artist, title)
      .then(lyrics => {
       
        this.setState({ lyrics });
      });
  };
  componentDidMount() {
    this.fetchLyrics();
  }

  render() {
    console.log(this.state.lyrics);
    return (
      <Fragment>
        <h1>{ this.state.lyrics }</h1>
      </Fragment>
    );
  }
}
