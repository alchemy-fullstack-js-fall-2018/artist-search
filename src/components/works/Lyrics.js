import React, { Component, Fragment } from 'react';
import { getLyrics } from '../../services/musicSearch';
import PropTypes from 'prop-types';

export default class Lyrics extends Component {
  static propTypes = {
    location: PropTypes.object.isRequired
  };

  state = {
    lyrics: '',
    artist: '',
    title: ''
  };

  componentDidMount() {
    const artist = this.props.location.pathname.slice(1).split('/')[1];
    const title = this.props.location.pathname.slice(1).split('/')[2];

    getLyrics(artist, title).then(result => {
      this.setState({ lyrics: result.lyrics, artist, title });
    });
  }

  render() {
    const { artist, title, lyrics } = this.state;

    return (
      <Fragment>
        <h3>{artist}</h3>
        <h4>{title}</h4>
        <p>{lyrics}</p>
      </Fragment>
    );
  }
}
