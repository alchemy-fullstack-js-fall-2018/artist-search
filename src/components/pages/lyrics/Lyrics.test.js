import React from 'react';
import { shallow } from 'enzyme';
import Lyrics from './Lyrics';

jest.mock('../../services/lyricSearch', () => {
  return { getLyrics: () => Promise.resolve({ lyrics: 'I believe in a thing called love' }) };
});

describe('Lyrics Component', () => {

  let wrapper;
  const location = { pathname: '/lyrics/Metallica/Atlas, Rise!' };
  beforeEach(() => {
    wrapper = shallow(
      <Lyrics
        location={location}
      />
    );
  });

  it('renders component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays artist, song and lyrics', () => {
    expect(wrapper.html()).toContain('Metallica');
    expect(wrapper.html()).toContain('Atlas, Rise');
    expect(wrapper.html()).toContain('I believe in a thing called love');
  });

});
