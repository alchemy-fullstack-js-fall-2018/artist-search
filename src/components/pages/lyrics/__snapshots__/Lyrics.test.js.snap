// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics Component renders component 1`] = `
<div>
  <h3>
    Metallica
     - 
    Atlas, Rise!
  </h3>
  <pre
    className="lyrics"
  >
    I believe in a thing called love
  </pre>
</div>
`;
